### PR TITLE
SL:88#179 add password attribute to meeting 

### PR DESCRIPTION
**Pull request by amalrails**
#179

### CODE REVIEW

**Review by kepstin — CHANGES_REQUESTED** [review]
Using a static moderator password is not the way to handle this - if the moderator password is not passed on the request, a *random per-meeting* moderator password must be generated.

The moderator password should be saved per-meeting in the meeting hash in redis.

I recommend having scalelite save the moderator password used in the received create call, if there was none passed then generate a new random one, and then pass that password on to the backend BBB server.

The moderator password should be saved in redis in the meeting model.

File: app/models/meeting.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # frozen_string_literal: true
 
 class Meeting < ApplicationRedisRecord
-  define_attribute_methods :id, :server_id
+  define_attribute_methods :id, :server_id, :password
```
Comment (kepstin) [inline app/models/meeting.rb:4]: Please call the attribute `moderator_pw` so we don't get it confused with the attendee password.

File: app/controllers/bigbluebutton_api_controller.rb [hunk 2]
```diff
@@ -147,7 +147,9 @@ def create
 
     # Create meeting in database
     logger.debug("Creating meeting #{params[:meetingID]} in database.")
-    meeting = Meeting.find_or_create_with_server(params[:meetingID], server)
+
+    moderator_pwd = params[:moderatorPW] || Rails.configuration.x.moderator_pwd
```
Comment (kepstin) [inline app/controllers/bigbluebutton_api_controller.rb:151]: Need to generate a new random password rather than using a static password.

**Review by kepstin — CHANGES_REQUESTED** [review]

File: app/controllers/bigbluebutton_api_controller.rb [hunk 3]
```diff
@@ -147,7 +147,10 @@ def create
 
     # Create meeting in database
     logger.debug("Creating meeting #{params[:meetingID]} in database.")
-    meeting = Meeting.find_or_create_with_server(params[:meetingID], server)
+
+    moderator_pwd = params[:moderatorPW] || SecureRandom.hex(6)
```
Comment (kepstin) [inline app/controllers/bigbluebutton_api_controller.rb:151]: We should probably try to match BBB itself as closely as possible in the random password generation: the relevant code is here: https://github.com/bigbluebutton/bigbluebutton/blob/363c80133895ca68c7825b463f8ca22df2d7f918/bbb-common-web/src/main/java/org/bigbluebutton/api/ParamsProcessorUtil.java#L647 (It generates a random alphanumeric 8-character string for the password using this function: http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/RandomStringUtils.html#randomAlphanumeric-int- )

Comment (kepstin) [reply app/controllers/bigbluebutton_api_controller.rb:151]: It looks like the `SecureRandom` class has this function builtin, you could so `SecureRandom.alphanumeric(8)`

File: app/controllers/bigbluebutton_api_controller.rb [hunk 4]
```diff
@@ -147,7 +147,10 @@ def create
 
     # Create meeting in database
     logger.debug("Creating meeting #{params[:meetingID]} in database.")
-    meeting = Meeting.find_or_create_with_server(params[:meetingID], server)
+
+    moderator_pwd = params[:moderatorPW] || SecureRandom.alphanumeric(8)
```
Comment (kepstin) [inline app/controllers/bigbluebutton_api_controller.rb:151]: We need to handle the case of a blank value (empty string) here. Rails provides a helper for this:
```suggestion
    moderator_pwd = params[:moderatorPW].presence || SecureRandom.alphanumeric(8)
```

**Review by kepstin — CHANGES_REQUESTED** [review]

**Review by jfederico — APPROVED** [review]

**Review by kepstin — APPROVED** [review]

**Pull request merged**